src/Cube/Heuristics/StateOne.hs

```diff
@@ -1,9 +1,112 @@
+    , right . right . right
+    , up . up . up
+    , down . down . down
+    ]
+
+{-
+ - Used to unfold a hashmap key to a list of integers.
+ - It just converts the numberic character at each index to its integer equivalent
+ -}
+unfoldToInt :: String -> Maybe (Int, String)
+unfoldToInt [] = Nothing
+unfoldToInt (x:xs) = Just (digitToInt x, xs)
+
+{-
+ - Used to fold a list of integers to a string.
+ - It just prepends the character value of the integer to the string.
+ - NOTE: Unfortunately this produces a string representation of the vector but in reverse order. This was specifically
+ - chosen because it is used extensively in the solve algorithm so reversing the vector is unnecessary overhead.
+ -}
+foldString :: String -> Int -> String
+foldString carry value = intToDigit value : carry
+
+{-
+ - Static lookup list. This is to be generated from a script ... TODO
+ -}
+heuristicList :: M.HashMap String Int
+heuristicList = M.singleton (replicate numEdges '0') 0
+
+{-
+ - Generate a list of edge states by applying all moves to the provided state.
+ -}
+nextStates :: Edge -> [Edge]
+nextStates edge = map (edge -:) moves
+
+{-
+ - Map the provided list of edge states to their corresponding heuristic index value.
+ -}
+-- TODO: combine this and the above function more elegantly
+heuristicIndices :: [Edge] -> [String]
+heuristicIndices = map $ V.foldl foldString "" . orien
+
+{-
+ - Set the list of keys to the value in the provided hashmap.
+ - This uses `seq` to ensure that duplicate keys already in the map are not overridden
+ -}
+setHashKeys :: [String] -> Int -> M.HashMap String Int -> M.HashMap String Int
+setHashKeys [] _ hash = hash
+-- TODO: this recursive application could be a source of distance distribution discrepancy
+setHashKeys (x:xs) dist hash = M.insertWith seq x dist $ setHashKeys xs dist hash
+
+{-
+ - Get a list of the keys for given value.
+ -}
+keysForDist :: Int -> M.HashMap String Int -> [String]
+keysForDist dist = M.keys . M.filter (dist ==)
+
+{-
+ - Ensure the given edge vector is the correct length.
+ - This will pad vector with zeros up to 11 then calculate the final element.
+ -}
+makeOrientation v = V.replicate padLength 0 V.++ v V.++ V.singleton modulus
+    where
+        currentLength = V.length v
+        padLength = numEdges - currentLength
+        modulus = V.sum v `mod` 2
 
 {-
- - This is used as a reducer for the edge orientation vector.
- - It treats the vector as a binary number and converts it to decimal using little-endian.
+ - Generate the full lookup map given a seed distance and map.
+ - The seed map should is expected to have the initial solved value and distance already set to allow for generating
+ - the states from.
  -}
-binaryFold carry index value = carry + value * 2 ^ index
+generateLookup :: Int -> M.HashMap String Int -> M.HashMap String Int
+generateLookup 8 hash = hash -- end state, the distance shouldnt exceed 7
+generateLookup dist hash = generateLookup (dist + 1) hash'
```

Comment:
TODO: convert this to a function wrapping a recursive implementation so the distance doesn't have to be passed to this function, it can just seed the recursive one. It might also be better if it didn't take a map at all or that is expected to be filled with at least one value already